packages/app-mobile/components/NoteEditor/MarkdownToolbar.tsx

```diff
@@ -0,0 +1,577 @@
+	const maxButtonsEachSide = Math.floor(
+		Math.min((maxButtons - 1) / 2, allButtonSpecs.length / 2)
+	) - 1;
+
+	const allButtonComponents: ButtonComponent[] = [];
+	let key = 0;
+	for (const spec of allButtonSpecs) {
+		key++;
+		allButtonComponents.push(
+			<ToolbarButton
+				key={key.toString()}
+				styles={props.styles}
+				spec={spec}
+			/>
+		);
+	}
+
+	const onToggleOverflowVisible = useCallback(() => {
+		AccessibilityInfo.announceForAccessibility(
+			!overflowButtonsVisible
+				? _('Opened toolbar overflow menu')
+				: _('Closed toolbar overflow menu')
+		);
+		setOverflowPopupVisible(!overflowButtonsVisible);
+	}, [overflowButtonsVisible]);
+
+	const toggleOverflowButton = (
+		<ToggleOverflowButton
+			key={(++key).toString()}
+			styles={props.styles}
+			overflowVisible={overflowButtonsVisible}
+			onToggleOverflowVisible={onToggleOverflowVisible}
+		/>
+	);
+
+	const mainButtons: ButtonComponent[] = [];
+	if (maxButtons < allButtonComponents.length) {
+		// We want the menu to look something like this:
+		// 			B I (…) 🔍 ⌨
+		// where (…) shows/hides overflow.
+		// Add from the left and right of [allButtonComponents] to ensure that
+		// the (…) button is in the center:
+		mainButtons.push(...allButtonComponents.slice(0, maxButtonsEachSide));
+		mainButtons.push(toggleOverflowButton);
+		mainButtons.push(...allButtonComponents.slice(-maxButtonsEachSide));
+	} else {
+		mainButtons.push(...allButtonComponents);
+	}
+
+	const mainButtonRow = (
+		<View style={props.styles.styleSheet.toolbarRow}>
+			{!overflowButtonsVisible ? mainButtons : null }
+		</View>
+	);
+
+	return (
+		<View style={props.styles.styleSheet.toolbarContainer}>
+			<ScrollView>
+				<OverflowRows
+					buttonGroups={props.buttons}
+					styles={props.styles}
+					visible={overflowButtonsVisible}
+					onToggleOverflow={onToggleOverflowVisible}
+				/>
+				{ !overflowButtonsVisible ? mainButtonRow : null }
+			</ScrollView>
+		</View>
+	);
+};
+
+interface MarkdownToolbarProps {
+	editorControl: EditorControl;
+	selectionState: SelectionFormatting;
+	searchState: SearchState;
+	editorSettings: EditorSettings;
+	style?: ViewStyle;
+}
+
+const MarkdownToolbar = (props: MarkdownToolbarProps) => {
+	const themeData = props.editorSettings.themeData;
+	const styles = useStyles(props.style, themeData);
+	const selState = props.selectionState;
+	const editorControl = props.editorControl;
+
+	const headerButtons: ButtonSpec[] = [];
+	for (let level = 1; level <= 5; level++) {
+		const active = selState.headerLevel === level;
+		let label;
+		if (!active) {
+			label = _('Create header level %d', level);
+		} else {
+			label = _('Remove level %d header', level);
+		}
+
+		headerButtons.push({
+			icon: `H${level}`,
+			accessibilityLabel: label,
+			active,
+
+			// This hook is executed the same number of times for each call
+			// of MarkdownToolbar.
+			// eslint-disable-next-line react-hooks/rules-of-hooks
+			onClick: useCallback(() => {
```

Comment:
Could you move this outside of the loop, and maybe pass the level to it from here? `onClick: onHeaderButtonClick({ level })` I'd rather we don't silence the rules of hooks.

Reply:
If extracted to a different function (both `useHeaderButtonClick` and `onHeaderButtonClick`), I get a different `rules-of-hooks` linter error. I think it makes more sense to silence the warning here without adding an extra function than silencing the warning within the wrapper function.